README.md

```diff
@@ -114,6 +114,27 @@ selinux_restore_dirs:
       - { login: '__default__', seuser: 'staff_u', serange: 's0-s0:c0.c1023', state: 'present' }
 ```
 
+#### Manage SELinux modules
+
+It's possible to maintain SELinux modules using `selinux_modules` variable which would contain a list of dictionaries, e.g.:
+
+```yaml
+    selinux_modules:
+      - { path: 'localmodule.pp', priority: '350', state: 'enabled' }
+      - { path: 'localmodule.cil', priority: '350', state: 'enabled' }
+      - { name: 'unconfineduser', priority: '100', state: 'disabled' }
+      - { name: 'unconfineduser', priority: '100', state: 'enabled' }
+      - { name: 'localmodule', priority: '350', state: 'absent' }
+```
+
+  - `path`: filename of a module to be installed, used for installing new modules
```

Comment:
filename on local/controller node, managed node, or both?

Reply:
It's a local file which is copied to a node and installed there.

Reply:
https://github.com/linux-system-roles/selinux/pull/56/commits/d7c9714abdf8aec5a69d53d791119c22cedd309e

Reply:
What types of files are supported?  Can I specify a `.te` file?  Is it that the file must be suitable for passing into `semodule -i` for the specified platform?
For example, as a sysadmin I want to specify policy for my RHEL7, RHEL8, and RHEL9 systems.  Do I need to create 3 different policy files?  How do I specify those in my inventory?

Reply:
> (either .cil or .pp) 

So currently no  .te as it's not SELinux module, it's part of SELinux module sources.

Unless a module uses types which are not defined in of the systems you should be able to use one .pp or .cil file for all. E.g. *zosremote* module exported on RHEL7 machine can be used on Fedora 33 just fine:

~~~
rhel7# semodule -E zosremote
rhel7# scp zosremote.pp fedora33:

fedora33# semodule -i zosremote.pp
libsemanage.semanage_direct_install_info: Overriding zosremote module at lower priority 100 with module at priority 400.
fedora33#  semodule -lfull | grep zosremote
400 zosremote         pp         
100 zosremote         pp     
~~~

It's not possible to generate a pp module on Fedora 33 and use it on RHEL7 as RHEL7 supports lower policy version

~~~
fedora33# semodule -E zosremote
fedora33# scp zosremote.pp rhel7:
rhel7# semodule -i zosremote.pp 
libsemanage.semanage_direct_install_info: Overriding zosremote module at lower priority 100 with module at priority 400.
libsemanage.semanage_pipe_data: Child process /usr/libexec/selinux/hll/pp failed with code: 255. (No such file or directory).
zosremote: libsepol.policydb_read: policydb module version 20 does not match my version range 4-19
zosremote: libsepol.sepol_module_package_read: invalid module in module package (at section 0)
zosremote: Failed to read policy package
libsemanage.semanage_direct_commit: Failed to compile hll files into cil files.
 (No such file or directory).
semodule:  Failed!
~~~

Export to CIL and installing .cil module can be used as a workaround:

~~~
fedora33# semodule -c -E zosremote
fedora33# scp zosremote.cil rhel7:

rhel7# semodule -i zosremote.cil 
libsemanage.semanage_direct_install_info: Overriding zosremote module at lower priority 100 with module at priority 400.

rhel7# semodule -lfull | grep zosremote
400 zosremote         cil         
100 zosremote         pp         
~~~

>

Reply:
In case when modules are really incompatible a sysadmin would need to pre-build the policy on each system and set role path variable according to the inventory I guess.

Reply:
Ok.  I think this should be documented in the README.  This was one of the use cases that Red Hat IT was asking about, how they would use this role instead of their custom selinux role.

Reply:
https://github.com/linux-system-roles/selinux/pull/56/commits/29c762b200ae10cab70d5719ee74ae858ccccda9